supervisor/addons/validate.py

```diff
@@ -336,7 +359,15 @@ def _migrate(config: dict[str, Any]):
         vol.Optional(ATTR_DEVICES): [str],
         vol.Optional(ATTR_UDEV, default=False): vol.Boolean(),
         vol.Optional(ATTR_TMPFS, default=False): vol.Boolean(),
-        vol.Optional(ATTR_MAP, default=list): [vol.Match(RE_VOLUME)],
+        vol.Optional(ATTR_MAP, default=list): [
+            vol.Schema(
+                {
+                    vol.Required("type"): str,
+                    vol.Optional("read_only"): bool,
```

Comment:
```suggestion
                    vol.Optional(ATTR_READ_ONLY, default=True): bool,
```
If this value is omitted, it should default to `True`. This keeps us from making `FolderMapping` instances with `None` for `read_only` and causing breakages here:
https://github.com/home-assistant/supervisor/blob/40bf7c081b4ad38fdfe661ce0ca88da2a770e09b/supervisor/docker/addon.py#L348

`ATTR_READ_ONLY` doesn't exist currently, it will need to be defined in `supervisor.const` or `supervisor.addons.const` so it can be reused.